content/master/concepts/packages.md

```diff
@@ -1,513 +1,489 @@
 metadata:
-  name: my-org-infra
+  name: platform-ref-aws
 spec:
-  crossplane:
-    version: ">=v1.0.0"
-  dependsOn:
-    - provider: xpkg.upbound.io/crossplane-contrib/provider-gcp
-      version: ">=v0.14.0"
+  revisionHistoryLimit: 10
+# Removed for brevity
 ```
 
-See all available fields in the [official documentation][configuration-docs].
-
-A Configuration package may also specify one or more of
-`CompositeResourceDefinition` and `Composition` types. These resources will be
-installed and will be solely owned by the Configuration package. No other
-package will be able to modify them.
+#### Configuration package pull policy
 
-The `spec.crossplane.version` field serves the same purpose that it does in a
-`Provider` package.
+Use a {{<hover label="pullpolicy" line="6">}}packagePullPolicy{{</hover>}} to
+define when Crossplane should download the Configuration package to the local
+Crossplane package cache.
 
-The `spec.dependsOn` field specifies packages that this package depends on. When
-installed, the package manager will ensure that all dependencies are present and
-have a valid version given the constraint. If a dependency is not installed, the
-package manager will install it at the latest version that fits within the
-provided constraints.
+The `packagePullPolicy` options are: 
+* `IfNotPresent` - (**default**) Only download the package if it isn't in the cache.
+* `Always` - Check for new packages every minute and download any matching
+  package that isn't in the cache.
+* `Never` - Never download the package. Packages are only installed from the
+  local package cache. 
 
-> Dependency resolution is a `beta` feature and depends on the `v1beta1`
-> [`Lock` API][lock-api].
+{{<hint "tip" >}}
+The Crossplane 
+{{<hover label="pullpolicy" line="6">}}packagePullPolicy{{</hover>}} works
+like the Kubernetes container image 
+[image pull policy](https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy).  
 
-For an example Configuration package, see [getting-started-with-gcp](https://github.com/crossplane/docs/tree/master/content/media/snippets/package/gcp).
+Crossplane supports the use of tags and package digest hashes like
+Kubernetes images. 
+{{< /hint >}}
 
-To build a Configuration package, navigate to the package root directory and
-execute the following command:
+For example, to `Always` download a given Configuration package use the 
+{{<hover label="pullpolicy" line="6">}}packagePullPolicy: Always{{</hover>}}
+configuration. 
 
-```
-crossplane build configuration
+```yaml {label="pullpolicy",copy-lines="6"}
+apiVersion: pkg.crossplane.io/v1
+kind: Configuration
+metadata:
+  name: platform-ref-aws
+spec:
+  packagePullPolicy: Always
+# Removed for brevity
 ```
 
-If the Provider package is valid, you will see a file with the `.xpkg`
-extension.
+#### Upgrade policy
 
-## Pushing a Package
+Crossplane automatically upgrades a Configuration the to the latest version 
+available in the package cache. 
```

Comment:
This doesn't have to be the case and could be misleading. 

Imagine the following:

1. Created a configuration package with `xpkg.upbound.io/upbound/platform-ref-aws:v1` 
2. XP downloaded `v1` to cache and installed it.
3. Updated `Configuration` spec.package to `xpkg.upbound.io/upbound/platform-ref-aws:v2` 
4. XP downloaded `v2` to the cache and upgraded to it with a new revision.
5. For some reason I decided to downgrade and change `Configuration` spec.package to `xpkg.upbound.io/upbound/platform-ref-aws:v1` 

At this state, while both `v1` and `v2` are in the cache, XP would activate `v1` as it is the desired state instead of _automatically upgrading to the latest version available in the package cache_.

Revision Activation Policy is about automatically marking the latest desired revision as active, and for Configuration packages, this doesn't have much user-facing impact in contrast to Providers where active revisions controller would process MRs. An active configuration revision would only own installed manifests (e.g. XRDs and Compositions`) as `controller=true`, which doesn't matter for the user.

Reply:
> If I change the XRD APIs between v1 and v2, wouldn't downgrading lose those APIs?

Yes, but I am not sure I understood the relation with the revisionActivationPolicy here 🤔 

You can read more details from the [design](https://github.com/crossplane/crossplane/blob/master/design/design-doc-packages-v2.md), but I couldn't find an easy answer there so let me try to explain:

Revision Activation Policy is all about the Package reconciler automatically marking a revision as `Active` or not. It doesn't impact which version is installed at any instant. `spec.package`  field of the package is authoritative on what version should be installed or not. 

Putting `revisionActivationPolicy` aside for a second, if you have the following `Configuration`, it will install `v1`:

```
apiVersion: pkg.crossplane.io/v1
kind: Configuration
metadata:
  name: platform-ref-aws
spec:
  package: xpkg.upbound.io/upbound/platform-ref-aws:v1
```

If you change it below, it will install `v2`:

```
apiVersion: pkg.crossplane.io/v1
kind: Configuration
metadata:
  name: platform-ref-aws
spec:
  package: xpkg.upbound.io/upbound/platform-ref-aws:v2
```

If you change it back to the previous spec, it will again install `v1`. The available versions in the package cache or in the registry doesn't have any impact here, you may have `v5` here or there, but the package manager will just install what you asked in the `Configuration` spec. It is your responsibility to have backward-compatible upgrades or supporting rollbacks, and package manager will just do whatever it is asked to do.

Reply:
So, what is `revisionActivationPolicy` then?

You can read more details from the [design](https://github.com/crossplane/crossplane/blob/master/design/design-doc-packages-v2.md), but I couldn't find an easy answer there, so let me try to explain:

Revision Activation Policy is all about the Package reconciler automatically marking a revision as `Active` or not. It doesn't impact which version is **installed** at any instant. `spec.package`  field of the package is authoritative on what version should be installed or not, as I explained above. 

So the question is, what is an active revision? A revision being active has the following meanings:

- Despite all package revisions owning the CRD, XRD, or Composition installed by the package, only the active one controls it with `controller: true` in the ownerRef. 
  - [This post](https://stackoverflow.com/a/65825463/4124416) explains the technical details well, but, in our context, we can think it as, which revision would be responsible for acting on the instances of those resources. 
  - As I stated above, this doesn't have a meaning for `Configuration` packages, since no controller is running for them different than `Provider`.
- Only the controller for the `Active` revision will be running to act on the instances. For a given `Provider` with `v1` and `v2` revisions installed where `v2` is marked as active, the provider controller for `v2` will be running and operating on the instances of the types that the provider installed, a.k.a. Managed Resources. 

When revisionActivationPolicy is automatic, the package manager will simply set the desired version (i.e. spec.package` in the `Provider) as active automatically.

As an example, I installed the following:

```
apiVersion: pkg.crossplane.io/v1
kind: Provider
metadata:
  name: provider-helm
spec:
  package: xpkg.upbound.io/crossplane-contrib/provider-helm:v0.14.0
  revisionActivationPolicy: Automatic
```

There is only 1 revision, which is for `v0.14.0`, and marked as `Active` by package manager (since we used Automatic policy)

```
❯ kubectl get providerrevisions.pkg.crossplane.io
NAME                         HEALTHY   REVISION   IMAGE                                                      STATE    DEP-FOUND   DEP-INSTALLED   AGE
provider-helm-ce18dd03e6e4   True      1          xpkg.upbound.io/crossplane-contrib/provider-helm:v0.14.0   Active                               50s
```

If I update the `spec.package` of `Provider` to `v0.15.0`:

```
❯ kubectl get providerrevisions.pkg.crossplane.io
NAME                         HEALTHY   REVISION   IMAGE                                                      STATE      DEP-FOUND   DEP-INSTALLED   AGE
provider-helm-503c3591121b   True      2          xpkg.upbound.io/crossplane-contrib/provider-helm:v0.15.0   Active                                 12s
provider-helm-ce18dd03e6e4   True      1          xpkg.upbound.io/crossplane-contrib/provider-helm:v0.14.0   Inactive                               2m40s
``` 

Not `v0.15.0` is marked as `Active` automatically. Meaning, that versions controller will process my MRs and I see the provider controller is running with its version:

```
❯ kubectl -n crossplane-system get pods provider-helm-503c3591121b-77fb968677-p6zp9 -o yaml | grep image
    image: xpkg.upbound.io/crossplane-contrib/provider-helm:v0.15.0
```

If I used `revisionActivationPolicy: Manual` in the above flow:

```
apiVersion: pkg.crossplane.io/v1
kind: Provider
metadata:
  name: provider-helm
spec:
  package: xpkg.upbound.io/crossplane-contrib/provider-helm:v0.14.0
  revisionActivationPolicy: Manual
``` 

It will install `v0.14.0` by creating a revision but won't mark it as `Active`, so no provider pod is running at all. No one will control, hence process the MRs of this provider. (For some reason, noticing that they were not initialized as `Inactive` either, but probably a small non-functional bug, we now they are not active)

```
❯ kubectl get providerrevisions.pkg.crossplane.io -w
NAME                         HEALTHY   REVISION   IMAGE                                                      STATE   DEP-FOUND   DEP-INSTALLED   AGE
provider-helm-ce18dd03e6e4   True      1          xpkg.upbound.io/crossplane-contrib/provider-helm:v0.14.0                                       13s

❯ kubectl -n crossplane-system get pods | grep provider-helm

```

Similarly, if I upgrade to `v0.15.0`, it will install it but still not run its controller since it will not be activated either.

We don't expect users to interact with this option much except for some advanced use cases. We only leveraged that during migration from monolith to family providers, to properly transfer ownership.